src/miniscript/types/extra_props.rs

```diff
@@ -121,11 +121,32 @@ pub struct ExtData {
     pub max_dissat_size: Option<(usize, usize)>,
     /// The timelock info about heightlocks and timelocks
     pub timelock_info: TimeLockInfo,
+    /// Maximum stack + alt stack size during satisfaction execution
+    /// This does **not** include initial witness elements. This element only captures
+    /// the additional elements that are pushed during execution.
+    pub exec_stack_elem_count_sat: Option<usize>,
+    /// Maximum stack + alt stack size during dissat execution
+    /// This does **not** include initial witness elements. This element only captures
+    /// the additional elements that are pushed during execution.
+    pub exec_stack_elem_count_dissat: Option<usize>,
 }
 
 impl Property for ExtData {
     fn sanity_checks(&self) {
-        //No sanity checks
+        fn xor<T>(a: Option<T>, b: Option<T>) -> bool {
+            match (a, b) {
+                (Some(_), Some(_)) | (None, None) => false,
+                _ => true,
+            }
+        }
+        debug_assert!(!xor(
+            self.stack_elem_count_sat,
+            self.exec_stack_elem_count_sat
+        ));
+        debug_assert!(!xor(
+            self.stack_elem_count_dissat,
+            self.exec_stack_elem_count_dissat
+        ));
```

Comment:
nit: this could probably be more clearly written as
```
debug_assert_eq!(self.stack_elem_count_stat.is_some(), self.exec_stack_elem_count_sat.is_some());
```
etc. without the `xor` helper.